WcaOnRails/app/helpers/competitions_helper.rb

```diff
@@ -1,2 +1,25 @@
 module CompetitionsHelper
+  def competition_message_for_user(competition, user)
+    message = ""
+    if registration = competition.registrations.find_by_user_id(user.id)
+      if registration.accepted?
+        message = "You are registered. "
+      else
+        message = "You are currently on the waiting list. "
+      end
+    end
+    if competition.isConfirmed
+      if competition.showAtAll
+        message += "This competition is confirmed and visible."
+      else
+        message += "This competition is confirmed but not visible."
+      end
```

Comment:
I very like the pattern with using `Array` and `join`.

About your solution, it's nice but this `how` variable looks ugly/meaningless (for me) so I'd create a variable to make `competition.showAtAll` shorter in usage and do this inline:
```ruby
  def competition_message_for_user(competition, user)
    messages = []
    if registration = competition.registrations.find_by_user_id(user.id)
      messages << "You are " + (registration.accepted? ? "registered." : "currently on the waiting list.")
    end
    visible = competition.showAtAll?
    if competition.isConfirmed
      messages << "This competition is confirmed #{visible ? 'and' : 'but not'} visible."
    else
      messages << "This competition is not confirmed #{visible ? 'but' : 'and not'} visible."
    end
    messages.join(' ')
  end
```

What do you think? =)